site/_posts/2020-08-07-locking-an-account-with-breached-password.md

```diff
@@ -0,0 +1,258 @@
+---
+layout: blog-post
+title: ! 'Breached Password Detection: How to Lock User Accounts with a Webhook'
+description: Webhooks let you take actions based on events in FusionAuth. For instance, if you've detected that someone tried logging in with a compromised password, you can immediately lock their account.
+author: Dan Moore
+image: blogs/breached-password-detection/how-to-enable-breached-password-detection-fusionauth.png
+category: blog
+tags: feature-breached-password-detection client-php feature-webhooks
+excerpt_separator: "<!--more-->"
+---
+
+Suppose you have an application to which access is so sensitive that if any user's password is found to be breached, the account should immediately be locked. This feature isn't built into FusionAuth. What can you do? 
```

Comment:
> This feature isn't built into FusionAuth. 

I don't know if I agree with this. If you turn on breach detection during login and force a password change, that is essentially the same as locking the account. 

I suppose the difference would be that we aren't (currently) providing an option to fail login, force a password change, and delete all "sessions' (refresh tokens). Perhaps we should add that as a configuration option. 

My only concern is to not minimize what we already offer. Could we say that we do allow you to force an immediate change to the password and block the login attempt - but if you want to completely disable the user, or lock their account ... read on?

Reply:
I can work on soft selling, but in my mind there is a difference between "user has to change their password and can login after that" and "user has account locked and can't gain access without admin unlocking the account".

Honestly, it's less about the particular action and more about how easy it is to extend fusionauth.